Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -568,143 +568,51 @@ void ArgsParser::add_positional_argument(Arg&& arg)
-{
-    Arg arg {
-        help_string,
-        name,
-        required == Required::Yes ? 1 : 0,
-        1,
-        [&value](char const* s) {
-            value = s;
-            return true;
-        }
-    };
-    add_positional_argument(move(arg));
-}
-
-void ArgsParser::add_positional_argument(String& value, char const* help_string, char const* name, Required required)
-{
-    Arg arg {
-        help_string,
-        name,
-        required == Required::Yes ? 1 : 0,
-        1,
-        [&value](char const* s) {
-            value = s;
-            return true;
-        }
-    };
-    add_positional_argument(move(arg));
-}
-
-void ArgsParser::add_positional_argument(StringView& value, char const* help_string, char const* name, Required required)
-{
-    Arg arg {
-        help_string,
-        name,
-        required == Required::Yes ? 1 : 0,
-        1,
-        [&value](char const* s) {
-            value = s;
-            return true;
-        }
-    };
-    add_positional_argument(move(arg));
-}
-
 void ArgsParser::add_positional_argument(int& value, char const* help_string, char const* name, Required required)
 {
-    Arg arg {
-        help_string,
-        name,
-        required == Required::Yes ? 1 : 0,
-        1,
+    add_positional_argument_helper(help_string, name, required,
         [&value](char const* s) {
             auto opt = StringView(s).to_int();
             value = opt.value_or(0);
             return opt.has_value();
-        }
-    };
-    add_positional_argument(move(arg));
+        });
 }
 
 void ArgsParser::add_positional_argument(unsigned& value, char const* help_string, char const* name, Required required)
 {
-    Arg arg {
-        help_string,
-        name,
-        required == Required::Yes ? 1 : 0,
-        1,
+    add_positional_argument_helper(help_string, name, required,
         [&value](char const* s) {
             auto opt = StringView(s).to_uint();
             value = opt.value_or(0);
             return opt.has_value();
-        }
-    };
-    add_positional_argument(move(arg));
+        });
 }
 
 void ArgsParser::add_positional_argument(double& value, char const* help_string, char const* name, Required required)
 {
-    Arg arg {
-        help_string,
-        name,
-        required == Required::Yes ? 1 : 0,
-        1,
+    add_positional_argument_helper(help_string, name, required,
         [&value](char const* s) {
             auto opt = convert_to_double(s);
             value = opt.value_or(0.0);
             return opt.has_value();
-        }
-    };
-    add_positional_argument(move(arg));
+        });
 }
 
-void ArgsParser::add_positional_argument(Vector<char const*>& values, char const* help_string, char const* name, Required required)
+template<bool IsContainer>
+void ArgsParser::add_positional_argument_helper(char const* help_string, char const* name, Required required, Function<bool(char const*)> accept_value)
```

Comment:
The template argument should not be named like the "concept" to avoid confusion. We name non-type template arguments like variables.

Reply:
Additionally, it looks like a bool-based enum would be much more readable here.

Reply:
I'll add that making that a normal bool argument won't hurt anything, or even passing whatever `INT_MAX` is from the caller